pkg/minikube/kubeconfig/kubeconfig.go

```diff
@@ -168,6 +168,13 @@ func writeToFile(config runtime.Object, configPath ...string) error {
 		fPath = configPath[0]
 	}
 
+	// checks permissions for ~/.kube/config
+	info, err := os.Stat(fPath)
+	m := info.Mode()
+	if m&(1<<2) == 0 {
+		return errors.Errorf("Could not write to file, give minikube permissions with:\nsudo chown -R $USER ~/.kube")
```

Comment:
I like the concept, but this PR has some issues that will need to be sorted first:

- Always check `err` before doing anything else
- This error message is not entirely correct: this function has not yet tested writing to a file.
- This check asserts that the owner of the file can write to it, not the current user. It is not accurate to suggest that `sudo` to fix an error that sudo can't fix, as it does not change the outcome of `info.Mode()`. 
- Suggesting `sudo` is going to be a bit weird for Windows users.

I suggest one of two approaches. The easiest option is adding a richer solution message to the err_map file: 

https://github.com/kubernetes/minikube/blob/72016f1012cdd6157a9df74e88b4fbe89ecc1e4f/pkg/minikube/problem/err_map.go#L394

If you want to  do things the more difficult way, I suggest:

- Try the write first. If it succeeds, return happiness.
- If the write fails, and the user does not own the file, return the error, with the suggestion that the file is not owned by the current user, and that they may need to change file ownership. https://stackoverflow.com/questions/58179647/getting-uid-and-gid-of-a-file has some guidance there.
- If the write fails, and the user does not have write permissions,  return the error with the suggestion that the file permissions do not allow write access.